src/classes/Contacts.cls

```diff
@@ -0,0 +1,748 @@
+
+    public Contacts(List<Contact> records) {
+        this(records, null, null);
+    }
+
+    public Contacts(List<Contact> records, List<Contact> oldList) {
+        this(records, oldList, null);
+    }
+
+    @TestVisible
+    private Contacts(List<Contact> records, List<Contact> oldList, HouseholdSettings
+            settings) {
+        super(records, Contact.SObjectType);
+        this.oldMap = oldList != null ?
+                new Map<Id, Contact>(oldList) : new Map<Id, Contact>();
+        this.settings = settings == null ? new HouseholdSettings() : settings;
+    }
+
+    public void onBeforeInsert() {
+        synchronizeNamingExclusionsFieldsBeforeInsert();
+    }
+
+    public void onBeforeUpdate() {
+        synchronizeNamingExclusionsFieldsBeforeUpdate();
+    }
+
+    public void onAfterInsert() {
+        processHouseholdsAfterInsert();
+    }
+
+    private void processHouseholdsAfterInsert() {
+        List<Contact> householdInserts = new List<Contact>();
+        List<Contact> contactEvaluations = new List<Contact>();
+        for (Contact con : (List<Contact>) getRecords()) {
+            if (needsHousehold(con)) {
+                householdInserts.add(con);
+            }
+            //If the user has chosen for only Contacts connected to Individual Accounts to have Households
+            //add them all to a list for evaluation
+            else if (needsEvaluation(con)) {
+                contactEvaluations.add(con);
+            } else if (needsHouseholdNameUpdated(con)) {
+                householdsNeedingNamesUpdated.add(householdIdFor(con));
+            }
+        }
+
+        if (householdsNeedingNamesUpdated.size() > 0) {
+            updateHouseholdNames(householdsNeedingNamesUpdated);
+        }
+
+        if (householdInserts.size() > 0) {
+            insertHouseholdsAfterInsert(householdInserts);
+        }
+
+        if (contactEvaluations.size() > 0) {
+            insertHouseholdsAfterInsert(
+                    getContactsNeedingHouseholds(contactEvaluations));
+        }
+    }
+
+    private Id householdIdFor(Contact con) {
+        return con.npo02__Household__c;
+    }
+
+    private Boolean needsHouseholdNameUpdated(Contact con) {
+        return householdIdFor(con) != null;
+    }
+
+    private Boolean needsEvaluation(Contact con) {
+        return householdIdFor(con) == null &&
+                settings.isAllIndividualsProcessor();
+    }
+
+    private Boolean needsHousehold(Contact con) {
+        return householdIdFor(con) == null
+                && settings.isAllProcessor();
+    }
+
+    public void buildOldHouseholdCollections() {
+        oldHouseholds.addAll(getOldHouseholdIds());
+        buildOldHouseholdSizeMap();
+    }
+
+    private void buildOldHouseholdSizeMap() {
+        for (AggregateResult a : getMemberCountForOldHouseholds()) {
+            oldHouseholdSizeById.put((id) a.get('oldhh'), (integer) a.get('membercount'));
```

Comment:
Let's declare constants for 'oldhh' and 'membercount'.

Reply:
I factored these down a bit for additional clarity but since the Aggregate query doesn't seem to be able to use the constant, I think it is ok to leave these as they are now.